src/components/dialogs/network-modifications/battery/modification/battery-modification-form.js

```diff
@@ -0,0 +1,134 @@
+    batteryToModify,
+    updatePreviousReactiveCapabilityCurveTable,
+    equipmentId,
+}) => {
+    const batteryIdField = (
+        <TextField
+            size="small"
+            fullWidth
+            label={'ID'}
+            value={equipmentId}
+            InputProps={{
+                readOnly: true,
+            }}
+            disabled
+            {...filledTextField}
+        />
+    );
+
+    const batteryNameField = (
+        <TextInput
+            name={EQUIPMENT_NAME}
+            label={'Name'}
+            formProps={filledTextField}
+            previousValue={batteryToModify?.name}
+            clearable={true}
+        />
+    );
+
+    const maximumActivePowerField = (
+        <FloatInput
+            name={MAXIMUM_ACTIVE_POWER}
+            label={'MaximumActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.maxP}
+            clearable={true}
+        />
+    );
+
+    const minimumActivePowerField = (
+        <FloatInput
+            name={MINIMUM_ACTIVE_POWER}
+            label={'MinimumActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.minP}
+            clearable={true}
+        />
+    );
+
+    const activePowerSetPointField = (
+        <FloatInput
+            name={ACTIVE_POWER_SET_POINT}
+            label={'ActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.targetP}
+            clearable={true}
+        />
+    );
+
+    const reactivePowerSetPointField = (
+        <FloatInput
+            name={REACTIVE_POWER_SET_POINT}
+            label={'ReactivePowerText'}
+            adornment={ReactivePowerAdornment}
+            previousValue={batteryToModify?.targetQ}
+            clearable={true}
+        />
+    );
+
+    return (
+        <>
+            <Grid container spacing={2}>
+                {gridItem(batteryIdField, 4)}
+                {gridItem(batteryNameField, 4)}
+            </Grid>
+            {/* Limits part */}
+            <GridSection title="Limits" />
+            <Grid container spacing={2}>
+                {gridItem(minimumActivePowerField, 4)}
+                {gridItem(maximumActivePowerField, 4)}
+            </Grid>
+            {/* Reactive limits part */}
+            <ReactiveLimitsForm
+                batteryInfos={batteryToModify}
+                updatePreviousReactiveCapabilityCurveTable={
+                    updatePreviousReactiveCapabilityCurveTable
+                }
+            />
+            {/* Set points part */}
+            <GridSection title="Setpoints" />
+            <Grid container spacing={2}>
+                {gridItem(activePowerSetPointField, 4)}
+                {gridItem(reactivePowerSetPointField, 4)}
+            </Grid>
+            <Grid container spacing={2} paddingTop={2}>
+                <FrequencyRegulation
+                    isbatteryToModify={true}
```

Comment:
the component FrequencyRegulation has no props named "isbatteryToModify" he has "isGeneratorModification"
to make this more flexible:
rename "isGeneratorModification" --> "isBatteryOrGeneratorModification" or "isEquipmentModification"
remove ""isbatteryToModify" and change it with perhaps isEquipmentModification={true}